src/data/update.ts

```diff
@@ -25,6 +26,8 @@ export const update = (api: VectorOperationsApi, namespace: string) => {
 
   return async (options: UpdateVectorOptions): Promise<void> => {
     validator(options);
+    options["setMetadata"] = options["metadata"];
```

Comment:
The way you added metadata to the type definition above implies that either `metadata` or `setMetadata` may be passed, but here in the implementation you are always overriding the value of setMetadata with the value of metadata (even if metadata was not passed). So there's a scenario where a person has called update passing a value in `setMetadata` that will then be overwritten as `undefined` before the request is made. E.g. something along the lines `client.index('my_index').update({id: '1', values: [0.1, 0.2, 0.3], setMetadata: { indexed: [ 'propertyName' ]}} )`. Even if we won't be showing the usage of setMetadata in the future, people who are migrating old code to use the new client could plausibly pass a value under the setMetadata key.

I would remove `setMetadata` from the type definition above to communicate our intention that `metadata` is what should be passed. And then I would wrap this assignment line `options["setMetadata"] = options["metadata"];` in an if statement that checks whether a value of setMetadata has been passed before overriding it. This is necessary because our typing is open so that if any new properties are added to the API in the future then they can immediately be used without updating the client.

Reply:
> This is necessary because our typing is open so that if any new properties are added to the API in the future then they can immediately be used without updating the client.

does this mean that we want to preserve the ability for `client.index('my_index').update({id: '1', values: [0.1, 0.2, 0.3], somethingTotallyNew: "foo" } )` to work with no client updates?  doesn't that fail because `somethingTotallyNew` won't validate?

Reply:
Currently calling this method with `somethingTotallynew: "foo"` would pass validation because our `UpdateVectorOptionsSchema` type definition does not include `{ additionalProperties: false }`. So it's a loose typing providing enforcement of the presence of required properties and the types of required and optional properties. Essentially, if you do not pass something we expect (e.g. you leave out the `id` property, or pass an `id` with incorrect type, the validator will reject it. But if you pass something new or unknown (or, unfortunately, if you misspell something optional), you won't get an error. 

The flexibility this affords to update properties that may not exist when this client is released is nice, but it comes at the cost of good feedback in some situations. For example, if you accidentally misspell the name of an optional property such as `metadata` as `meatdata`, you will not get feedback from the validator.

It's a tough call between the benefits of stricter typing vs the flexibility to tolerate certain kinds of changes in the API. For now we've stayed on the side of flexibility, but if you think we should take a more rigid stance then we can talk about it more.

Reply:
that makes sense, and also means my [revised implementation](https://github.com/pinecone-io/pinecone-ts-client/pull/83/files/c72c5f58eb16335f98b4ef67ca797fc8daf8eb89#diff-6910af72b7db376da51a11a1638b35d8910dcb07662cd625fb17293acf34bf65L28) is going to break that principle because it assumes we know all of the request params.  i'll take another pass at it!

Reply:
following up based on an offline conversation -- that principle doesn't hold because our API requests have typed parameters, so the [implementation here](https://github.com/pinecone-io/pinecone-ts-client/pull/83/files/c72c5f58eb16335f98b4ef67ca797fc8daf8eb89#diff-6910af72b7db376da51a11a1638b35d8910dcb07662cd625fb17293acf34bf65L28) will work.